codice/ddf
# DDF-3493 Extract duplicated logic for SP metadata parser into its own class

#### What does this PR do?
Pulls logic for parsing service provider metadata into its own class
#### Who is reviewing it? 
@blen-desta 
@Lambeaux 
#### Select relevant component teams: 
#### Choose 2 committers to review/merge the PR. 
(please choose ONLY two committers from below, delete the rest)
@coyotesqrl
@rzwiefel
#### How should this be tested? (List steps with links to updated documentation)
We need to test the parser on multiple metadata source types: **url**, **file uri**, **xml inline block**

To test **url** source:
1. Just log into ddf. Url is the default scheme so if logging in works then the url parsing works.

To test **file uri** source:
1. Create a new file anywhere in your DDF distribution (I chose the `/etc` directory). Remember your file's name and absolute path.
2. In your address bar type `https://${hostname}:${port}/services/saml/sso/metadata`. Copy/paste the xml from that webpage into your new file.
3. Log into the admin console and navigate to `Security -> Configuration -> IdP Server`
4. Delete everything in the `SP Metadata` field and replace it with the absolute path to your file (including your file in the path) prepended with `file:`.
So if the absolute path to my file is `C:\users\hot-guy\myfile.xml` I would replace `SP Metadata` with: `file:C:\users\hot-guy\myfile.xml`.
5. Save changes, restart your browser, and try to log into DDF. A successful login means it worked!

To test **xml inline block** source:
1. Log into the admin console and navigate to `Security -> Configuration -> IdP Server`
2. Delete everything in the `SP Metadata` field and replace it directly with the xml you copied to your file in step 2 of the **file uri** section. There should just be xml in the `SP Metadata` field when you are done.
3. Save changes, restart your browser, and try to log into DDF. A successful login means it worked!

#### Any background context you want to provide?
#### What are the relevant tickets?
[DDF-3493](https://codice.atlassian.net/browse/DDF-3493)
#### Screenshots (if appropriate)
#### Checklist:

#### Notes on Review Process
Please see [Notes on Review Process](https://codice.atlassian.net/wiki/spaces/DDF/pages/71946981/Pull+Request+Guidelines) for further guidance on requirements for merging and abbreviated reviews. 

#### Review Comment Legend:
- ✏️ (Pencil) This comment is a nitpick or style suggestion, no action required for approval. This comment should provide a suggestion either as an in line code snippet or a gist. 
- ❓ (Question Mark) This comment is to gain a clearer understanding of design or code choices, clarification is required but action may not be necessary for approval.
- ❗ (Exclamation Mark) This comment is critical and requires clarification or action before approval.
